flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/typeinfo/RowSerializer.scala

```diff
@@ -102,20 +119,39 @@ class RowSerializer(fieldSerializers: Array[TypeSerializer[Any]])
     val len = fieldSerializers.length
 
     val result = new Row(len)
-    var i = 0
-    while (i < len) {
-      result.setField(i, fieldSerializers(i).deserialize(source))
-      i += 1
+
+    var index = 0
+    while (index < len) {
+      val isNull: Boolean = source.readBoolean()
+      if (isNull) {
+        result.setField(index, null)
+      } else {
+        val serializer: TypeSerializer[Any] = fieldSerializers(index)
+        result.setField(index, serializer.deserialize(source))
+      }
+      index += 1
     }
     result
   }
 
+  private final val booleanSerializer = new BooleanSerializer()
+
   override def copy(source: DataInputView, target: DataOutputView): Unit = {
     val len = fieldSerializers.length
     var i = 0
     while (i < len) {
+      booleanSerializer.copy(source, target)
```

Comment:
I think it would be easier to do
```
target.writeBoolean(source.readBoolean())
```
here, instead of going through the extra abstraction of the BooleanSerializer.